src/app/components/secret-key/mnemonic-key/mnemonic-input-field.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useState } from 'react';
+
+import { TextField } from '@radix-ui/themes';
+import { useField } from 'formik';
+import { css } from 'leather-styles/css';
+import { FlexProps, styled } from 'leather-styles/jsx';
+
+import { useIsFieldDirty } from '@app/common/form-utils';
+
+interface InputFieldProps extends FlexProps {
+  dataTestId?: string;
+  name: string;
+  value: string;
+  onPaste: (e: React.ClipboardEvent<HTMLInputElement>) => void;
+  onUpdateWord: (word: string) => void;
+  hasError?: boolean;
+  wordlist: string[];
+}
+export function InputField({ dataTestId, name, onPaste, onChange, value }: InputFieldProps) {
+  const [field, meta] = useField(name);
+  const [isFocused, setIsFocused] = useState(false);
+  const isDirty = useIsFieldDirty(name);
+
+  const getBorderColour = useCallback(() => {
+    if (isDirty && meta.error) {
+      // FIXME #4130: show a red border on error - couldn't get it working
+      return 'colors.error';
+    }
+    if (isFocused) {
+      return 'accent.text-primary';
+    }
+    return 'none';
+  }, [isDirty, isFocused, meta.error]);
+
+  return (
+    <TextField.Root
```

Comment:
Adding these notes here as perhaps we will create a wrapper for this if we decide to use the radix field `TextField` app-wide